exchange/src/main/java/org/geneontology/gocam/exchange/Helper.java

```diff
@@ -151,4 +158,112 @@ public static void writeOntologyAsObo(String outfile, OWLOntology ont) throws OW
 		ont.getOWLOntologyManager().setOntologyFormat(ont, new OBODocumentFormat());	
 		ont.getOWLOntologyManager().saveOntology(ont,outf);
 	}
+	
+	public static Map<String, String> parseGPI(String gpiFile) throws IOException {
+		Map<String, String> idLookup = new HashMap<String, String>();
+		
+		//neo needs e.g.
+		//http://identifiers.org/sgd/S000001024
+		//we get e.g.
+		//YeastCyc YIL155C-MONOMER
+		//we get the mapping from a YeastCyc GPI file downloaded from http://sgd-archive.yeastgenome.org/curation/literature/ 
+		//e.g. http://sgd-archive.yeastgenome.org/curation/literature/gp_information.559292_sgd.gpi.gz 
+		BufferedReader reader = new BufferedReader(new FileReader(gpiFile));
+		String line = reader.readLine();
+		while(line!=null) {
+			if(line.startsWith("!")) {
+				line = reader.readLine();
+			}else {
+				String[] cols = line.split("	");
+				String yeastcyc = cols[4];
+				Set<String> yeastcyc_ids = new HashSet<String>();
+				if(yeastcyc.contains("|")) {
+					String[] ids = yeastcyc.split("\\|");
+					for(String id : ids) {
+						yeastcyc_ids.add(id);
+					}
+				}else {
+					yeastcyc_ids.add(yeastcyc);
+				}
+				if(cols[0].contentEquals("ComplexPortal")) {
+					for(String yeastacc : yeastcyc_ids) {
+						idLookup.put(yeastacc, "http://purl.obolibrary.org/obo/ComplexPortal_"+cols[1]);
```

Comment:
We fixed these ComplexPortal ids; there might be a mismatch somewhere: https://github.com/geneontology/neo/pull/81

Reply:
Oh I get it now, just needed to change to the new prefix. Fixed in a1bb852.

Reply:
Sorry, my comment was not very informative!